AEPMessaging/Tests/UnitTests/MessagingTests.swift

```diff
@@ -1191,6 +1191,48 @@ class MessagingTests: XCTestCase {
         XCTAssertEqual(MOCK_PUSH_TOKEN, getDispatchedEventPushToken(event: pushTokenEvent))
     }
 
+    func testMultiplePushTokenSync_whenTokenMatchesAndOutsideSyncTimeout_forceSyncTrue() {
+        // setup
+        let delay = UInt32(1500000) // 1.5 seconds
+        let mockConfig = [EXPERIENCE_CLOUD_ORG: MOCK_EXP_ORG_ID, MessagingConstants.SharedState.Configuration.PUSH_FORCE_SYNC: true] as [String : Any]
+        let eventData: [String: Any] = [MessagingConstants.Event.Data.Key.PUSH_IDENTIFIER: MOCK_PUSH_TOKEN]
+
+        let event = Event(name: "handleProcessEvent", type: EventType.genericIdentity, source: EventSource.requestContent, data: eventData)
+        mockRuntime.simulateSharedState(for: MessagingConstants.SharedState.Configuration.NAME, data: (value: mockConfig, status: SharedStateStatus.set))
+        mockRuntime.simulateXDMSharedState(for: MessagingConstants.SharedState.EdgeIdentity.NAME, data: (value: SampleEdgeIdentityState, status: SharedStateStatus.set))
+        usleep(delay)
```

Comment:
nit: For these two sleeps, add a comment stating something to the effect of "sleeping to add delta between event timestamps".